rotkehlchen/chain/ethereum/modules/cowswap/decoder.py

```diff
@@ -0,0 +1,236 @@
+        other_events = []  # other events such as approvals
+        for event in decoded_events:
+            if (
+                event.event_type == HistoryEventType.RECEIVE and
+                event.counterparty == GPV2_SETTLEMENT_ADDRESS
+            ):
+                receive_events[(event.asset, event.balance.amount)] = event.location_label  # type: ignore[assignment]  # location label is not None here  # noqa: E501
+            elif (
+                event.event_type != HistoryEventType.SPEND or
+                event.counterparty != GPV2_SETTLEMENT_ADDRESS
+            ):
+                other_events.append(event)
+
+        relevant_trades: list[tuple[TradeData, ChecksumEvmAddress]] = []  # trades and traders addresses  # noqa: E501
+        for trade in trades:
+            trader_address = receive_events.get((trade.to_asset, trade.to_amount))
+            if trader_address is not None:
+                relevant_trades.append((trade, trader_address))
+
+        return relevant_trades, other_events
+
+    def _aggregator_post_decoding(
+            self,
+            transaction: EvmTransaction,
+            decoded_events: list[HistoryBaseEntry],
+            all_logs: list[EvmTxReceiptLog],
+    ) -> list[HistoryBaseEntry]:
+        if transaction.to_address != GPV2_SETTLEMENT_ADDRESS:
+            return decoded_events
+
+        all_trades = self._find_trades(all_logs)
+        relevant_trades, other_events = self._detect_relevant_trades(all_trades, decoded_events)
+
+        new_events = []
+        for trade, trader_address in relevant_trades:
+            # Generate a spend event
+            new_events.append(self.base.make_event_next_index(
+                tx_hash=transaction.tx_hash,
+                timestamp=transaction.timestamp,
+                event_type=HistoryEventType.TRADE,
+                event_subtype=HistoryEventSubType.SPEND,
+                asset=trade.from_asset,
+                balance=Balance(amount=trade.from_amount),
+                location_label=trader_address,
+                notes=f'Swap {trade.from_amount} {trade.from_asset.symbol} in {CPT_COWSWAP}',
+                counterparty=CPT_COWSWAP,
+            ))
+
+            # Generate a receive event
+            new_events.append(self.base.make_event_next_index(
+                tx_hash=transaction.tx_hash,
+                timestamp=transaction.timestamp,
+                event_type=HistoryEventType.TRADE,
+                event_subtype=HistoryEventSubType.RECEIVE,
+                asset=trade.to_asset,
+                balance=Balance(amount=trade.to_amount),
+                location_label=trader_address,
+                notes=f'Receive {trade.to_amount} {trade.to_asset.symbol} as the result of a swap in {CPT_COWSWAP}',  # noqa: E501
+                counterparty=CPT_COWSWAP,
+            ))
+
+        for event in other_events:  # Make sure that the indices here don't overlap with trades
```

Comment:
this is not right. This will move all events **after** the trades. Makes no sense. Even the gas burning. There is an order that should be preserved. If there was somehting like a gas fee, or an approval before you should keep it before.